### ISSUE TITLE
Remove the Equipment Sync Step (and additional steps)

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#1982]
**[Mingle Card: 2214](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/2214)**
This is a technical card and should be covered by Tarantula.
------------------------------------------------------------

This card has expanded to removing multiple sync steps: equipment, equipment planning, contacts, crew planning

Acceptance Criteria

_All of the equipment pickers in mobile should still work.  It should also use the Search Result Layout for Equipment when it pulls up the dialog._

Related Stories
---------------

This can be removed because of #2131.

Analysis

DP - this is partially done.  The sync step is removed and the equipment picker is working with the dynamic equipment lookup data.  See https://github.com/dpitre/alpine-mobile/tree/remove-equipment

 Steps left:

*   Swap out the Equipment Result for the regular picker.
*   Add a multi-select option to the picker.
*   Add a filter to the picker so it only shows equipment from the current price book and filter out scheduled stuff

\------------- Everything below here is not the Plan

The equipment will now be filtered by the sharing rules and permission sets for fields.

Currently, the equipment step includes all the equipment for the users configured office set.  It then includes any extraordinary pieces of equipment based on equipment planning.

This change would entirely remove our support for all of this and replace it with sharing rules and field permission configurations.  Documentations and exchange of knowledge with the imps would be necessary.

Simply removing the step and removing equipment from the query-handler exclusion list should just work.  However, story points does not take into consideration education time.

After a thorough review of the sharing rules it boils down to these strategies:

*   Since equipment is used differently than the way it is tracked, add a distribution\_region picklist to the object to base the sharing rules off of.  Since there is a limit of 50 sharing rules/object, this would work for less complicated orgs where the distribution of equipment sync data is uncomplicated.
*   For trickier orgs, this could be used to control the sharing: [http://www.salesforce.com/us/developer/docs/apexcode/Content/apex\_bulk\_sharing\_creating\_with\_apex.htm](http://www.salesforce.com/us/developer/docs/apexcode/Content/apex_bulk_sharing_creating_with_apex.htm)

**The Gruesome Part**

The picker for adding new ticket items base on equipment is a complete and utter mess for the following reasons:

*   Equipment is being mixed with Equipment Schedule and then split apart again when they are shown in the UI.
*   Filtering on the add by equipment screen is currently hard coded.  There is no reason it should remain this way.  
*   We should be able to show the standard picker with custom filtering base on available price book items.
*   The add sections in the ticket item grid is one huge brick of code.  It needs to be split up before this card is attempted.
*   Because of these other things, it may be the case that contacts, scheduled contacts, scheduled equipment should be done at the same time.

**Dev Plan**

*   Sync down equipment as a generic lookup, but KEEP the current equipment sync step. This allows us to proceed with making all lookups generic without getting blocked by (or breaking) the add-ticket-item functionality. This means that in the best case scenario, the equipment sync step is removed when the add-to-ticket functionality is cleaned up. Worst case scenario, we end up syncing 2 sets of equipment for an iteration while we complete the refactor.

Tasks

{{table query: SELECT Number, Name, Owner, 'Task Status' WHERE Type = Task and Story = THIS CARD}}

Defects
-------

{{table query: SELECT Number, Name, Owner, 'Defect Status' WHERE Type = Defect and 'Related Story' = THIS CARD}}

Test Plan
---------

**Comment by githubbob42** [comment]
### Feature: #1909 Dynamic Lookup Handling ([Mingle](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/2138))
